File: Data Structures & Algorithms/products-of-array-discluding-self/submission-13.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res(nums.size(), 1);
        for(int i = 1; i < nums.size(); i++){
            res[i] = nums[i-1] * res[i-1];
        }
        int pre = 1;
        for(int j = nums.size()-1; j >=0; j--){
            res[j] *= pre;
            pre *= nums[j];
        }
        return res; 
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-13.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        curr = []
        res = []
        nums.sort()
        def dfs(i, tar):
            if tar == 0:
                res.append(curr.copy())
                

            for j in range(i, len(nums)):
                if tar < nums[j]:
                    return
                curr.append(nums[j])
                dfs(j, tar - nums[j])
                curr.pop()

        dfs(0, target)
        return res